services/workers/src/test/java/org/sagebionetworks/table/worker/DatasetIntegrationTest.java

```diff
@@ -151,14 +154,53 @@ public void testQueryDataset()
 							.setValues(Arrays.asList("v-3")), rows.get(2));
 				}, MAX_WAIT);
 	}
+	
+	// Reproduce PLFM-7025
+	@Test
+	public void testQueryDatasetWithFileSize()
+			throws AssertionError, AsynchJobFailedException, DatastoreException, InterruptedException {
+
+		int numberOfVersions = 2;
+		
+		FileEntity fileOne = createFileWithMultipleVersions(1, stringColumn.getName(), numberOfVersions);
+		FileEntity fileTwo = createFileWithMultipleVersions(2, stringColumn.getName(), numberOfVersions);
```

Comment:
files and versions of files should have distinct sizes (4 different sizes altogether?)